source/core/slang-downstream-compiler.h

```diff
@@ -18,6 +18,8 @@ namespace Slang
 
 struct DownstreamDiagnostic
 {
+    typedef DownstreamDiagnostic ThisType;
+
     enum class Type
```

Comment:
Unrelated to this PR, but we should probably name this "severity" eventually, to match what we use for our own diagnostics. It might also be nice to share some subset of those fundamental types between these cases, but I guess the issue here is that we are in `core` here rather than `slang`.

Reply:
(I'm referring to the `enum class Type` here, and not the `ThisType` declaration that the PR added, just to be clear.)

(And part of the context here is my ongoing effort to avoid using the term "type" to refer to anything that is not actually or arguably a type in the PL sense of the term)